src/EditorFeatures/VisualBasicTest/ConvertLinq/ConvertLinqQueryToLinqMethodTests.vb

```diff
@@ -0,0 +1,117 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.CodeRefactorings
+Imports Microsoft.CodeAnalysis.VisualBasic.ConvertLinq
+
+Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.CodeActions.ConvertLinq
+
+    Public Class ConvertLinqQueryToLinqMethodTests
+        Inherits AbstractVisualBasicCodeActionTest
+
+        Protected Overrides Function CreateCodeRefactoringProvider(workspace As Workspace, parameters As TestParameters) As CodeRefactoringProvider
+            Return New VisualBasicConvertLinqQueryToLinqMethodProvider()
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertLinq)>
+        Public Async Function Conversion_WhereOrderByTrivialSelect() As Task
+            Await Test(
+"From num In New Integer() {0, 1, 2} Where num Mod 2 = 0 Order By num Select num",
+"New Integer() {0, 1, 2}.Where(Function(num) num Mod 2 = 0).OrderBy(Function(num) num)")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertLinq)>
+        Public Async Function Conversion_WhereSelect() As Task
+            Await Test(
+"From x in New Integer() {0, 1, 2} Where x<5 Select x* x",
+"New Integer() {0, 1, 2}.Where(Function(x) x < 5).Select(Function(x) x * x)")
+        End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertLinq)>
+        Public Async Function Conversion_MultipleFrom() As Task
+            Await Test(
+"From x In New Integer() {0} From y In New Integer() {1} From z In New Integer() {2} Where x + y + z < 5 Select x * x",
+"New Integer() {0}.SelectMany(Function(x) New Integer() {1}, Function(x, y) New With {x, y
+       }).SelectMany(Function(VBIt) New Integer() {2}, Function(VBIt1, z) New With {VBIt1, z
```

Comment:
>VBIt1 [](start = 64, length = 5)

We need to figure out what to do with names. For example, what will we do if "z" was actually named "VBIt1".